### PR TITLE
test: add consensus-critical justification and finalization test vectors

### PR DESCRIPTION
**Pull request by uink45**
## 🗒️ Description

Add 10 new state transition test vectors to `test_justification.py` exercising consensus-critical edge cases for justification and finalization:

- **Exact 2/3 threshold boundary:** Verifies that 4-of-6 validators, hitting the exact `>=` boundary, is sufficient to justify a target.
- **Below-threshold rejection:** Verifies that 3-of-6 validators, falling below the two-thirds threshold, does not advance justification.
- **Vote accumulation across blocks:** Verifies that pending votes from earlier blocks are preserved and eventually cross the threshold when new validators contribute in a later block.
- **Repeated validator dedup across blocks:** Verifies that the same validators voting twice for the same target across separate blocks do not inflate the tally.
- **Finalization on next justifiable step:** Verifies that justifying the immediately next justifiable slot after the source finalizes the source checkpoint.
- **No finalization with intermediate justifiable slot:** Verifies that justifying a distant target advances the justified checkpoint but does not finalize the source when intermediate justifiable slots exist in between.
- **Mid-block finalized-slot visibility:** Verifies that a second attestation within the same block can see the finalized-slot update caused by the first attestation during mid-block processing.
- **Pronic boundary acceptance:** Verifies that a target at pronic distance (delta 6 = 2×3) from the finalized slot is accepted for justification.
- **Non-justifiable boundary rejection:** Verifies that a target at non-justifiable distance (delta 7) from the finalized slot is rejected despite supermajority support.
- **Square boundary acceptance:** Verifies that a target at square distance (delta 9 = 3²) from the finalized slot is accepted for justification.

## 🔗 Related Issues or PRs

N/A

## ✅ Checklist

- [X] Ran `tox` checks to avoid unnecessary CI fails:
    ```console
    uvx tox
    ```
- [X] Considered adding appropriate tests for the changes.
- [ ] Considered updating the online docs in the [./docs/](/leanEthereum/leanSpec/tree/main/docs/) directory.

### CODE REVIEW

File: tests/consensus/devnet/state_transition/test_justification.py [hunk 1]
```diff
@@ -61,3 +61,645 @@ def test_supermajority_attestations_justify_block(
...
+                            ValidatorIndex(4),
+                            ValidatorIndex(5),
+                        ],
+                        slot=Slot(4),
+                        target_slot=Slot(2),
+                        target_root_label="block_2",
+                    ),
+                ],
+            ),
+        ],
```
Comment (tcoratger) [inline tests/consensus/devnet/state_transition/test_justification.py:286]: Here I don't think that we are testing the duplication thing properly because even if dedup was broken, `latest_justified_slot` would still be 2 because validators 2, 3, 4, 5 voted for this separately.

To truly isolate dedup, we need:
  - 4 validators, validators 0,1 vote in block A, then validators 0,1 vote again in block B (same target). Unique = 2/4. Double-counted = 4/4.
  - 3*2 >= 2*4 → 6 >= 8 → False (correct, no justify)
  - 3*4 >= 2*4 → 12 >= 8 → True (broken dedup would justify)
  - Assert `latest_justified_slot=0`

Comment (uink45) [reply tests/consensus/devnet/state_transition/test_justification.py:286]: Thanks for the feedback. I have updated the PR to account for this.
